airflow/security/kerberos.py

```diff
@@ -80,7 +82,7 @@ def perform_krb181_workaround():
     ret = subprocess.call(cmdv, close_fds=True)
 
     if ret != 0:
-        principal = "%s/%s" % (
+        principal = args.principal or "%s/%s" % (
```

Comment:
I'm not sure this change is right - I'd have guessed that args.principal would be used to override the principal in the config and then add to it?

Reply:
We don't override config here. It is used only to prepare a command - https://github.com/apache/incubator-airflow/blob/3dc9e4021786e2c2416956f83a1f19ea1052ff3b/airflow/security/kerberos.py#L46

Reply:
This is a change in behaviour.

Previously:

```
principal == "the-actual-principal/my-fqdn"
```

After if I specify the arg:

```
principal == "the-actual-principal"
```

i.e. if arg.principal is provided the format is never called, and I think it should be?

Reply:
oh, I understand what you talking about, I think you right, yes. My mistake.